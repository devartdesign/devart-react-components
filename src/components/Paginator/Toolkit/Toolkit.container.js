import { connect } from 'react-redux';

// import { paginatorState } from '../../../../state-mgmt';
import Toolkit from './Toolkit';

export const mapStateToProps = state => ({
  search: state.paginator.search,
  filters: state.blog.filters
});

// export const mapDispatchToProps = dispatch => ({
//   onSearch: text => dispatch(paginatorState.actions.onSearch(text)),
// });

export default connect(mapStateToProps, null)(Toolkit);
