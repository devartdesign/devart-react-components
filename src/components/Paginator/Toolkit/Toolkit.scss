.paginator__toolkit {
  display: flex;
  // height: 35px;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  margin-bottom: 30px;

  &__item {
    margin-left: 20px;
  }
}