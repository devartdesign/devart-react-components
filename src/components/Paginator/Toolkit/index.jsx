export { default } from './Toolkit.container';
