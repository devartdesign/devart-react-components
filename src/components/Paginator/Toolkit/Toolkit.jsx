import React from 'react';
import PropTypes from 'prop-types';

import Search from './Search';
import Filter from './Filter';

import './Toolkit.scss';

export default class Toolkit extends React.PureComponent {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    filterOptions: PropTypes.array,
    hideColumnOptions: PropTypes.array,
    paginationProps: PropTypes.object,
    disableFilter: PropTypes.bool,
    disableHideColumns: PropTypes.bool,
    disableSearch: PropTypes.bool,
    search: PropTypes.string,
    filters: PropTypes.object
  };

  static defaultProps = {
    filterOptions: [],
    hideColumnOptions: [],
    paginationProps: {},
    disableFilter: false,
    disableHideColumns: false,
    disableSearch: false,
    search: '',
    filters: {},
  };

  state = {
    filters: [],
    hideColumns: []
  };

  componentDidUpdate(prevProps) {
    // esto tiene que ir en un side effect..
    const { filters, search, paginationProps, onChange } = this.props;
    if (prevProps.search !== search) {
      onChange({
        ...paginationProps,
        type: 'search',
        search,
        filters,
        hideColumns: this.state.hideColumns
      });
    }
    if (prevProps.filters !== filters) {
      onChange({
        ...paginationProps,
        type: 'filters',
        search,
        filters,
        hideColumns: this.state.hideColumns
      });
    }
  }

  handleChange = (response) => {
    this.setState({
      [response.name]: response.value
    }, () => {
      this.props.onChange({
        ...this.props.paginationProps,
        type: response.name,
        search: this.props.search,
        filters: this.state.filters,
        hideColumns: this.state.hideColumns
      });
    });
  };

  render() {
    const {
      hideColumnOptions,
      filterOptions,
      disableFilter,
      disableHideColumns,
      disableSearch
    } = this.props;

    return (
      <div className="paginator__toolkit">
        {!disableSearch && (
          <div className="paginator__toolkit__item">
            <Search
              onSearch={this.handleChange}
              value={this.props.search}
            />
          </div>
        )}
        {!disableFilter && (
          <div className="paginator__toolkit__item">
            <Filter
              name="filters"
              label="Filters"
              onSelect={this.handleChange}
              options={filterOptions}
              selectedOptions={this.state.filters}
            />
          </div>
        )}
        {!disableHideColumns && (
          <div className="paginator__toolkit__item">
            <Filter
              name="hideColumns"
              label="Hide Columns"
              onSelect={this.handleChange}
              options={hideColumnOptions}
              selectedOptions={this.state.hideColumns}
              width={140}
            />
          </div>
        )}
      </div>
    );
  }
}
