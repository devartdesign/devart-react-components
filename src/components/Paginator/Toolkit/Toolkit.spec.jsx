import React from 'react';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';

import { getInitialState } from '../../../../mocks/store';

import Toolkit from './Toolkit';

describe('<PaginatorToolkit />', () => {
  let wrapper;
  const defaultProps = {
    onChange: jest.fn(),
    filterOptions: [],
    paginationProps: {
      currentPage: 1,
      pageLimit: 2,
      totalRecords: 2,
      totalPages: 2
    }
  };

  describe('render', () => {
    beforeEach(() => {
      const mockStore = configureStore();
      const store = mockStore(getInitialState());
      wrapper = mount(
        <Provider store={store}>
          <Toolkit {...defaultProps} />
        </Provider>
      );
    });

    it('should render a .paginator_toolkit wrapper with a <Search> and a <Filter> components', () => {
      expect(renderer.create(wrapper)
        .toJSON())
        .toMatchSnapshot();
    });
  });

  describe('handleChange', () => {
    beforeEach(() => {
      wrapper = shallow(<Toolkit {...defaultProps} />);
    });

    it('should call onChange callback from Props', () => {
      wrapper.instance().handleChange({ name: 'search', value: 'some text to be searched' });
      expect(defaultProps.onChange).toHaveBeenCalledWith({
        currentPage: 1,
        pageLimit: 2,
        totalRecords: 2,
        totalPages: 2,
        search: '',
        filters: [],
        hideColumns: [],
        type: 'search'
      });
    });
  });
});
