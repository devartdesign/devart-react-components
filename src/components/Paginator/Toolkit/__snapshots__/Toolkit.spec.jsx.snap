// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginatorToolkit /> render should render a .paginator_toolkit wrapper with a <Search> and a <Filter> components 1`] = `
<div
  className="paginator__toolkit"
>
  <div
    className="paginator__toolkit__item"
  >
    <div
      className="Search-search-1"
    >
      <div
        className="Search-searchIcon-2"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-5"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </div>
      <div
        className="MuiInputBase-root-27 MuiInput-root-14 Search-inputRoot-3 MuiInput-underline-18"
        onClick={[Function]}
      >
        <input
          className="MuiInputBase-input-37 MuiInput-input-22 Search-inputInput-4"
          name="search"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Searchâ€¦"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="paginator__toolkit__item"
  >
    <div
      className="MuiFormControl-root-44"
      style={
        Object {
          "width": 100,
        }
      }
    >
      <label
        className="MuiFormLabel-root-59 MuiInputLabel-root-48 MuiInputLabel-formControl-53 MuiInputLabel-animated-56"
        data-shrink={false}
        htmlFor="select-multiple-checkbox"
        style={
          Object {
            "left": 4,
            "top": -10,
          }
        }
      >
        Filters
      </label>
      <div
        className="MuiInputBase-root-27 MuiInput-root-14 MuiInput-underline-18 MuiInputBase-formControl-28 MuiInput-formControl-15"
        onClick={[Function]}
        style={
          Object {
            "margin": 5,
          }
        }
      >
        <div
          className="MuiSelect-root-66"
        >
          <div
            aria-haspopup="true"
            aria-pressed="false"
            className="MuiSelect-select-67 MuiSelect-selectMenu-70 MuiInputBase-input-37 MuiInput-input-22"
            id="select-filters"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            id="select-multiple-checkbox"
            name="filters"
            placeholder="Filters"
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-5 MuiSelect-icon-72"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="paginator__toolkit__item"
  >
    <div
      className="MuiFormControl-root-44"
      style={
        Object {
          "width": 140,
        }
      }
    >
      <label
        className="MuiFormLabel-root-59 MuiInputLabel-root-48 MuiInputLabel-formControl-53 MuiInputLabel-animated-56"
        data-shrink={false}
        htmlFor="select-multiple-checkbox"
        style={
          Object {
            "left": 4,
            "top": -10,
          }
        }
      >
        Hide Columns
      </label>
      <div
        className="MuiInputBase-root-27 MuiInput-root-14 MuiInput-underline-18 MuiInputBase-formControl-28 MuiInput-formControl-15"
        onClick={[Function]}
        style={
          Object {
            "margin": 5,
          }
        }
      >
        <div
          className="MuiSelect-root-66"
        >
          <div
            aria-haspopup="true"
            aria-pressed="false"
            className="MuiSelect-select-67 MuiSelect-selectMenu-70 MuiInputBase-input-37 MuiInput-input-22"
            id="select-hideColumns"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            id="select-multiple-checkbox"
            name="hideColumns"
            placeholder="Filters"
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-5 MuiSelect-icon-72"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
